docs/source/app-dev/command-deduplication.rst

```diff
@@ -0,0 +1,364 @@
+
+   - Set the timeout (gRPC deadline) to the expected submission processing time (Command Service) or submission hand-off time (Command Submission Service).
+
+     The **submission processing time** is time between when the application sends off a submission to the :ref:`Command Service <command-service>` and when it receives (synchronously, unless it times out) the acceptance or rejection.
+     The **submission hand-off time** is the time between when the application sends off a submission to the :ref:`Command Submission Service <command-submission-service>` and when it obtains a synchronous response for this gRPC call.
+     After the RPC timeout, the application considers the submission as lost and enters a retry loop.
+     This timeout is typically much shorter than the deduplication duration.
+
+   .. _dedup-bounded-step-await:
+   
+#. Wait until the RPC call returns a response.
+   
+   - Status codes other than ``OK`` should be handled according to :ref:`error handling <dedup-bounded-error-handling>`.
+
+   - When you use the :ref:`Command Service <command-service>` and the response carries the status code ``OK``, the ledger change took place.
+     You can report success.
+     
+   - When you use the :ref:`Command Submission Service <command-submission-service>`,
+     subscribe with the :ref:`Command Completion Service <command-submission-service>` for completions for ``actAs`` from ``OFF1`` (exclusive) until you see a completion event for the change ID and the submission ID chosen in :ref:`Step 3 <dedup-bounded-step-submit>`.
+     If the completion’s status is ``OK``, the ledger change took place and you can report success.
+     Other status codes should be handled according to :ref:`error handling <dedup-bounded-error-handling>`.
+   
+     This step needs no timeout as the :ref:`Command Submission Service <command-submission-service>` acknowledges a submission only if there will eventually be a completion event, unless relevant parts of the system become permanently unavailable.
+
+
+.. _dedup-bounded-error-handling:
+
+Error handling
+--------------
+
+Error handling is needed when the status code of the command submission RPC call or in the :ref:`in the completion event <com.daml.ledger.api.v1Completion.status>` is not ``OK``.
+The following table lists appropriate reactions by status code (written as ``STATUS_CODE``) and error code (written in capital letters with a link to the error code documentation).
+Fields in the error metadata are written as ``field`` in lowercase letters.
+
+.. list-table:: Command deduplication error handling with known processing time bound
+   :widths: 10 50
+   :header-rows: 1
+
+   - * Error condition
+     
+     * Reaction
+
+       
+   - * ``DEADLINE_EXCEEDED``
+     
+     * Consider the submission lost.
+       
+       Retry from :ref:`Step 2 <dedup-bounded-step-offset>`, "obtaining the completion offset ``OFF1``", and possibly increase the timeout.
```

Comment:
The exact text "obtaining the completion offset" is not part of `Step 2`, so this may be confusing. Maybe summarize the step in parenthesis?
```suggestion
       Retry from :ref:`Step 2 <dedup-bounded-step-offset>` (i.e., obtaining the completion offset ``OFF1``), and possibly increase the timeout.
```

Reply:
I had parentheses before and Carrie suggested to replace them by commas. As a compromise, I'll remove the quotes.